include/aws/http/private/proxy_impl.h

```diff
@@ -103,7 +103,10 @@ struct aws_http_proxy_user_data {
     struct aws_socket_options original_socket_options;
     bool original_manual_window_management;
     size_t original_initial_window_size;
+    bool prior_knowledge_http2;
     struct aws_http1_connection_options original_http1_options;
+    struct aws_http2_connection_options original_http2_options; /* allocated with userdata */
```

Comment:
not sure why `/* allocated with userdata */` comment is on `original_http2_options` which is a by-value thing

Reply:
The options has setting array, which need to be allocated from heap. The comment was for that, will clarify it more.